.travis.yml

```diff
@@ -47,8 +48,19 @@ jobs:
       before_deploy:
         - docker login -u "${DOCKER_USER}" -p "${DOCKER_PASS}"
       deploy:
-        provider: script
-        on:
-          tags: true
-        script: /bin/bash scripts/docker_push.sh "${TRAVIS_TAG}" latest
-...
\ No newline at end of file
+        - provider: script
+          on:
+            tags: true
+          script: /bin/bash scripts/docker_push.sh "${TRAVIS_TAG}" latest
+    - stage: 'pypi deployment'
+      #if: type = pull_request OR tag IS present  ###uncomment this line in order to work for specific cases similar to ""stage: 'Build docker image'"" 
```

Comment:
We don't need the same mechanism as for the Docker build. The reason is that for Docker, we build the image on pull requests and on tags. The image is then pushed only on tags. This enables us to check for problems in the image building process when a pull request is made.

For PyPI, the deployment provider both builds the package and uploads it to PyPI, so we are going to do this only on tags. So the only condition which is necessary is the `on: tags: true` statement in the deployment provider configuration.

Reply:
I have added what you want.  :)

Reply:
Alright. I have modified it.

Reply:
Please have a look @aperrin66

Reply:
It looks all right to me, the `if: tag IS present` statement might be redundant but it doesn't hurt :)
Can you just remove the empty line at the end of the file?

Reply:
It is done. I removed them. and push it  on Commit 9660ec9